src/tools/web/kdb.js

```diff
@@ -0,0 +1,116 @@
+// kdb.js - small library to access Elektra's kdb via node.js
+const { exec } = require('child_process')
+
+// constants
+const ERR_KEY_NOT_FOUND = 'Did not find key'
+
+// KDBError
+function KDBError (message) {
+    this.name = 'KDBError'
+    this.message = message || ''
+}
+KDBError.prototype = Error.prototype
+
+// remove newline from the end of a string
+const trimNewline = (str) =>
+  str.substring(0, str.length - 1)
+
+// execute a script while catching and parsing errors
+const safeExec = (script) => new Promise((resolve, reject) =>
+  exec(script, (err, stdout, stderr) => {
+    if (err) {
+      const errors = err.message.split('\n')
+      // ignore error if it's "key not found"
+      if (!(errors.length > 1 && errors[1] === ERR_KEY_NOT_FOUND)) {
+        return reject(err)
+      }
+    }
+    if (stderr) {
+      if (stderr !== ERR_KEY_NOT_FOUND) {
+        return reject(new KDBError(stderr))
+      } else {
+        return resolve() // no key found, return no value
+      }
+    }
+    const result = trimNewline(stdout)
+    if (!result) {
+      return resolve() // empty result, return no value
+    }
+    return resolve(result)
+  })
+)
+
+// escape strings by surrounding them with ""
+const escapeValues = (template, ...values) =>
+  template.reduce((acc, part, i) => {
+    let val = values[i - 1]
+    if (typeof val === 'string') val = `"${val}"`
+    return acc + val + part
+  })
+
+const ELEKTRA_VERSION_REGEX = /KDB_VERSION\:\ ([0-9]+\.[0-9]+\.[0-9]+)\n/
+
+// get Elektra version
+const version = () =>
+  safeExec(`kdb --version`)
+    .then(output => { // parse result
+      const matches = ELEKTRA_VERSION_REGEX.exec(output)
+      if (!matches) {
+        throw new Error('invalid version: ' + output)
+      }
+      return matches
+    })
+    .then(matches => matches.length > 1 && matches[1]) // select version from matches
+    .then(fullVersion => {
+      const splitVersions = fullVersion.split('.')
+      return {
+        version: fullVersion,
+        major: Number(splitVersions[0]),
+        minor: Number(splitVersions[1]),
+        micro: Number(splitVersions[2])
+      }
+    })
+
+// list available paths under a given `path`
+const ls = (path) =>
+  safeExec(escapeValues`kdb ls ${path}`)
+    .then(stdout => stdout && stdout.split('\n'))
+
+// get value from given `path`
+const get = (path) =>
+  safeExec(escapeValues`kdb get ${path}`)
+
+// set value at given `path`
+const set = (path, value) =>
+  safeExec(escapeValues`kdb set ${path} ${value}`)
+
+// remove value at given `path`
+const rm = (path) =>
+  safeExec(escapeValues`kdb rm ${path}`)
+
+// export javascript object from given `path`
+const _export = (path) =>
```

Comment:
It still doesn't work.

Here is a Vagrantfile that matches the environment in which i tested your pullrequest (HASH is your pullrequests hash)
```ruby
Vagrant.configure("2") do |config|
  config.vm.box = "ubuntu/xenial64"

  config.vm.provision "shell", inline: <<-SHELL
    apt-get update
    apt-get install -qq -y curl build-essential cmake pkg-config libyajl-dev

    curl -sL https://deb.nodesource.com/setup_6.x | bash -
    apt-get install -qq -y nodejs

    HASH='0823edac81b0a33a53ab321a7cdf1aa456dd80ed'
    curl https://codeload.github.com/ElektraInitiative/libelektra/tar.gz/${HASH} | tar xz --transform s/libelektra-${HASH}/libelektra/
    mkdir libelektra/build && cd $_ && cmake .. && make && make install
    echo "/usr/local/lib" > /etc/ld.so.conf.d/usrlocal.conf
    ldconfig
  SHELL
end
```
If you start nodejs and execute
```node
child_process.exec('kdb export "user/sw/elektra/web/#0/current/clusters" yajl', (err, stdout, stderr) => console.log(stderr)); '';
```
this error is thrown. Although
`kdb export "user/sw/elektra/web/#0/current/clusters" yajl` works in the commandline.

I found this error, because i can't open clusters because of this error.